src/DetailsView/components/details-view-command-bar.tsx

```diff
@@ -10,23 +12,32 @@ import { FeatureFlagStoreData } from '../../common/types/store-data/feature-flag
 import { TabStoreData } from '../../common/types/store-data/tab-store-data';
 import { DetailsViewActionMessageCreator } from '../actions/details-view-action-message-creator';
 import { DetailsRightPanelConfiguration } from './details-view-right-panel';
-import { ReportExportComponent, ReportExportComponentDeps } from './report-export-component';
+import { ReportExportComponent, ReportExportComponentDeps, ReportExportComponentProps } from './report-export-component';
 import { StartOverDropdown } from './start-over-dropdown';
 
 export type DetailsViewCommandBarDeps = ReportExportComponentDeps & {
     getCurrentDate: () => Date;
     reportGenerator: ReportGenerator;
+    getDateFromTimestamp: (timestamp: string) => Date;
 };
 
+export type CommandBarProps = Omit<DetailsViewCommandBarProps, 'renderStartOver'>;
+
+export type ReportExportComponentPropertyConverter = (props: CommandBarProps) => ReportExportComponentProps;
+
 export interface DetailsViewCommandBarProps {
     deps: DetailsViewCommandBarDeps;
     featureFlagStoreData: FeatureFlagStoreData;
     tabStoreData: TabStoreData;
     actionMessageCreator: DetailsViewActionMessageCreator;
     assessmentStoreData: AssessmentStoreData;
     assessmentsProvider: AssessmentsProvider;
-    renderExportAndStartOver: boolean;
+    reportExportComponentProps: ReportExportComponentProps;
+    renderStartOver: boolean;
     rightPanelConfiguration: DetailsRightPanelConfiguration;
+    visualizationScanResultData: VisualizationScanResultData;
+    cardsViewData: CardsViewModel;
+    reportExportComponentPropertyFactory?: ReportExportComponentPropertyConverter;
```

Comment:
In what case does the `DetailsViewCommandBar `need both the factory and the ```reportExportComponentProps```?

As far as I can see it's not being used, so maybe just missed on clean up?

Reply:
Ah, I see you're using the same interface for exposing the Assessment and FastPass command bars: I would instead create  a new interface for that purpose that includes this factory and not make it optional.

Reply:
Moved to deps. As discussed offline, the single interface is intentional

Reply:
As discussed, ran into problems with deps. Now picking up the property factories from the switcherNavConfiguration property